### ISSUE TITLE
decorate and retrieve `@examples` calls from `testthat/`

### ISSUE THREAD

**Issue by maxheld83** [dataheld/elf#12]
another (third?) alternative idea for the existing `source_pef()`-clumsiness, after #11 #4 have failed:

It'd be nice to have some magic markup/decoration *inside* the testthat that can then be called from roxygen decorators, like so (absolute pseudocode):

`testthat/test-foo.R`:

```r
res <- {#' @exampleFromTestthat bar_1
  bar_fun(1, 2)
}
test_that("lorem", expect_true(res))
```

`R/foo.R`:

```r
#' Foo function
#' @reuseExamplesFromTestthat bar_1
...
```

**Comment by maxheld83** [comment]
however this is still quite convoluted, and I'm not sure this can be built cleanly, or at all.

I'm guessing this would require some fancy work with environments, because I'm not sure of any other way to mark *some* part of the testthat code as reusable, while also still evaluating it in terms of testthat.
(By contrast, the stuff behind roxygen2 or plumber decorations *doesn't* get evaluated in the normal course of evaluation of a file; they're *commments*.)
So this might get fairly deep into the language/parser ... I don't even know.

It's probably a bridge too far for a relatively modest increase in convenience,
while it's also still somewhat ham-fisted/un-idiomatic.

Leaving this here on backlog for reference.
